tools/projmgr/docs/Manual/Overview.md

```diff
@@ -0,0 +1,1317 @@
+- Libraries, source, and header files that are not configurable (and need no modification) are stored in the directory of the Software Component (typically part of CMSIS_Pack_ROOT) and get included directly from this location into the project.
+- An Include Path to the header files of the Software Component is added to the C/C++ Compiler control string.
+
+### PLM of configuration files
+
+Configurable source and header files have a version information that is required during Project Lifetime Management (PLM) of a project. The version number is important when the underlying software pack changes and provides a newer configuration file version.
+
+Depending on the PLM status of the application, the `projmgr` performs for configuration files the following operation:
+
+1. **Add** a software component for the first time: the related config file is copied twice into the related `RTE` project directory.  The first copy can be modified by the user with the parameters for the user application. The second copy is an unmodified hidden backup file that is appended with the version information.
+
+    **Example:** A configuration file `ConfigFile.h` at version `1.2.0` is copied:
+
+    ```c
+    ./RTE/component_class/ConfigFile.h           // user editable configuration file
+    ./RTE/component_class/.ConfigFile.h-1.2.0    // hidden backup used for version comparison
+    ```
+    
+    The `projmgr` outputs a user notification to indicate that files are added:
+
+    ```text
+    ./RTE/component_class/ConfigFile.h -  info: component 'name' added configuration file version '1.2.0'
+    ```
+
+2. **Upgrade** (or downgrade) a software component: if the version of the hidden backup is identical, no operation is performed. If the version differs, the new configuration file is copied with appended version information.
+
+    **Example:** after configuration file `ConfigFile.h` to version `1.3.0` the directory contains these files:
+
+    ```c
+    ./RTE/component_class/ConfigFile.h           // user editable configuration file
+    ./RTE/component_class/ConfigFile.h-1.3.0     // user editable configuration file
+    ./RTE/component_class/.ConfigFile.h-1.2.0    // hidden backup used for version comparison
+    ```
+
+    The `projmgr` outputs a user notification to indicate that configuration files have changed:
+
+    ```text
+    ./RTE/component_class/ConfigFile.h - warning: component 'name' upgrade for configuration file version '1.3.0' added, but file inactive
+    ```
+
+3. **User action to complete upgrade**: The user has now several options (outside of `projmgr`) to merge the configuration file information.  A potential way could be to use a 3-way merge utility. After merging the configuration file, the hidden backup should be deleted and the unmodified new version should become the hidden backup.  The previous configuration file may be stored as backup as shown below.
```

Comment:
Can't we specify a preferred merge tool in some user settings so that projMgr invokes the merge tool with the appropriate 3 files ?